Userland/Utilities/ls.cpp

```diff
@@ -142,6 +143,22 @@ int main(int argc, char** argv)
     for (size_t i = 0; i < paths.size(); i++) {
         auto path = paths[i];
 
+        if (flag_recursive && Core::File::is_directory(path)) {
```

Comment:
it's not immediately obvious to me whether this would spin on circular symlinks, have you tested that?
```
$ mkdir -p foo/bar
$ cd foo/bar; ln -s .. baz
$ cd ../..
$ ls -R
```

Should not spin.

Reply:
You're correct -- it was spinning on circular symlinks, and I was completely unaware. Based on my testing, I think I fixed that now by implementing `LibCore::File::is_link()` and using it within `ls`.